packetnetworking/conftest.py

```diff
@@ -159,3 +159,21 @@ def _mockit(i, *args, **kwargs):
         return mock.patch(i.__module__ + "." + i.__name__, *args, **kwargs)
 
     return _mockit
+
+
+@pytest.fixture
+def make_interfaces_dhcp_metadata(metadata):
+    def _make_interfaces_dhcp_metadata(overrides=None):
+        md = metadata(overrides or {})
```

Comment:
You shouldn't need to call metadata anymore since the passed argument is the already generated metadata.

Changing `overrides=None` to `md` and removing this line would (also removing the metadata fixture) would be more proper.